### DISCUSSION TITLE
Fresh SvelteKit deployment crashes, produces “Serverless Function has crashed”

### DISCUSSION THREAD

**Discussion by renefournier** [vercel/vercel#6750] [Help] [answered]
Hi everyone, 

Trying my first vanilla SvelteKit deployment to Vercel and encountering a Function error:
```This Serverless Function has crashed.

Your connection is working correctly.

Vercel is working correctly.

500: INTERNAL_SERVER_ERROR
Code: FUNCTION_INVOCATION_FAILED
ID: pdx1::mbzm9-1632272724809-36ae24cc9e6c
```

When I look at logs for Functions, I see:
```[GET] /
19:10:27:89
2021-09-22T01:10:27.959Z    undefined    ERROR    TypeError: Cannot read property 'paths' of undefined
    at init (/var/task/index.js:3248:22)
    at Object.<anonymous> (/var/task/index.js:4071:1)
    at Module._compile (internal/modules/cjs/loader.js:1072:14)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1101:10)
    at Module.load (internal/modules/cjs/loader.js:937:32)
    at Function.Module._load (internal/modules/cjs/loader.js:778:12)
    at ModuleWrap.<anonymous> (internal/modules/esm/translators.js:199:29)
    at ModuleJob.run (internal/modules/esm/module_job.js:170:25)
    at async Loader.import (internal/modules/esm/loader.js:178:24)
    at async getListener (/var/task/___vc/__launcher.js:33:11)
RequestId: 256fd573-2e02-459a-a205-299d3ffe6c6a Error: Runtime exited with error: exit status 1
Runtime.ExitError
```
Now, the app runs fine locally, and in fact I'm not doing any API calls. It's just a website with a couple pages. Anyone know where I might look further?

**Comment by leerob** [comment]
Sorry about this. Did you start with the [SvelteKit template](https://vercel.com/new/clone?s=https%3A%2F%2Fgithub.com%2Fvercel%2Fvercel%2Ftree%2Fmain%2Fexamples%2Fsveltekit&template=sveltekit&id=67753070&b=main&from=templates) by chance?

**Reply by renefournier** [reply]
Hi @leerob,

I started with the SvelteKit template from https://kit.svelte.dev and then `npm i` the vercel adapter and added it to `svelte.config.js`. From the link you shared, it looks like I should have started with the Vercel-blessed template, right?

**Reply by leerob** [reply]
The Vercel template started from there as well but sets up the adapter for you. It's possible they've shipped a breaking change, as it's still in beta - does that template work for you by chance?

**Reply by leerob** [reply]
I was just able to deploy the template linked above successfully 👍

**Reply by renefournier** [reply]
Oh very good! Thanks so much, I will try the Vercel template tomorrow, confident it will get me there. 'Night!

**Reply by renefournier** [reply]
Hi, I managed to deploy from the template, too. Thanks. One thing I notice that is missing in `npm run dev` is that the HMR doesn't work. That is, if I make a change to, say, `index.svelte`, press Save—the browser doesn't automatically update. I have to `CMD-R`. Anyone know how we might fix that?

**Reply by renefournier** [reply]
Also, Vercel says that pushing to `main` branch will prompt a redeploy, but that is not happening for me.

**Reply by leerob** [reply]
I'm not able to reproduce - HMR is working for me. On pushing to `main`, does the Vercel GitHub integration have access to that repository? You might have only access for one specific repo - make sure it's allowed 👍

**Reply by renefournier** [reply]
Re: HMR, hmm, strange. Re: pushing to main, I'm using BitBucket and I don't know if Vercel has access... I mean, when I created the deployment and imported the repo, the initial deployment worked, but any subsequent pushes to main don't go anywhere, and my BitBucket says “Vercel Git author must have access...”.

**Reply by leerob** [reply]
Can you link your project please? Based on that message, it sounds like your BitBucket account isn't part of the Vercel team.

**Reply by renefournier** [reply]
Hi @leerob. It's here: https://www4-r2-creative.vercel.app/

So, I started over again, selected the SvelteKit template, deployed... OK. Then made a commit in BitBucket editor, pushed... and Vercel auto-deployed. So, working. Not sure what I did wrong before... Will git clone to my machine, make another change, push to main and see what happens.

**Reply by renefournier** [reply]
Ack, stopped working again. Is there like a one-hour timeout on the repo auth thing?

**Comment by renefournier** [comment, marked as answer]
Original issue resolved by using Vercel template
